### PR TITLE
PR for 48 - Fix crash after error and rotate

### LINKED ISSUE #48: Crash when rotating event list after error
"when I loaded the page initially I had no network connection, and the error was shown. After rotating the device, it crashed"

```
12-20 18:20:53.168  8796  8796 E AndroidRuntime: FATAL EXCEPTION: main
12-20 18:20:53.168  8796  8796 E AndroidRuntime: Process: com.schedjoules.eventdiscovery.demo, PID: 8796
12-20 18:20:53.168  8796  8796 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke interface method 'boolean com.schedjoules.client.eventsdiscovery.ResultPage.isFirstPage()' on a null object reference
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at com.schedjoules.eventdiscovery.eventlist.itemsprovider.ScrollDirection$1.hasComingPageQuery(ScrollDirection.java:42)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at com.schedjoules.eventdiscovery.eventlist.itemsprovider.EventListItemsProviderImpl.queueComingPage(EventListItemsProviderImpl.java:151)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at com.schedjoules.eventdiscovery.eventlist.itemsprovider.EventListItemsProviderImpl.onScrolledCloseToEdge(EventListItemsProviderImpl.java:176)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at com.schedjoules.eventdiscovery.eventlist.view.EdgeReachScrollListener.onScrollStateChanged(EdgeReachScrollListener.java:69)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.support.v7.widget.RecyclerView.dispatchOnScrollStateChanged(RecyclerView.java:4426)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.support.v7.widget.RecyclerView.setScrollState(RecyclerView.java:1334)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.support.v7.widget.RecyclerView.onTouchEvent(RecyclerView.java:2756)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.View.dispatchTouchEvent(View.java:9415)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2660)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2666)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2318)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2666)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2318)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2666)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2318)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2666)
12-20 18:20:53.168  8796  8796 E AndroidRuntime:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2318)
```

### PR DESCRIPTION
**Pull request by lemonboston**
fixes #48 

@dmfs please review.

### CODE REVIEW

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/async/DiscardingPreExecuteSafeAsyncTask.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+
+/**
+ * Improved {@link AsyncTask} with a callback (kept with WeakReference) that delivers both success and failed results
+ * along with the input params of the task, and with a {@link DiscardCheck} callback that is called from multiple points
+ * in the execution to enable the client to discard/cancel the task. There is also a onPreExecute callback.
+ *
+ * @author Gabor Keszthelyi
+ */
+public abstract class DiscardingPreExecuteSafeAsyncTask<TASK_PARAM, EXECUTE_PARAM, PROGRESS, TASK_RESULT> extends AsyncTask<EXECUTE_PARAM, PROGRESS, SafeAsyncTaskResult<TASK_RESULT>>
```
Comment (dmfs) [inline eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/async/DiscardingPreExecuteSafeAsyncTask.java:34]: Is this class ever used anywhere?

**Review by dmfs — APPROVED** [review]
Approving this. It seems to work and an unused class is not a big issue at this point.

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/async/DiscardingPreExecuteSafeAsyncTask.java (hunk 1 above)
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/async/DiscardingPreExecuteSafeAsyncTask.java:34]: No, I just wanted to save this version as well as part of a kind of general AsyncTask mini-library that might be extracted later from current `framework.async` package.
But we can remove it as well.

Comment (dmfs) [reply eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/async/DiscardingPreExecuteSafeAsyncTask.java:34]: Just keep it for now.

**Pull request merged**